### PR TITLE
Allow a delegate to be run when a request times out #108

### PR DESCRIPTION
**Pull request by kbrimble**

### CODE REVIEW

**Comment by kbrimble** [comment]
Not sure why the build of this PR has failed. It has nothing to do with the changes that I made.

**Comment by scottoffen** [comment]
Master has some failing unit tests. I'll update that today from my branch, and you can rebase off of master. That should fix the issue.

File: src/Grapevine/Client/RestClient.cs [hunk 1]
```diff
@@ -66,8 +66,11 @@ public class RestClient : IRestClient
         public CookieContainer Cookies { get; }
         public ICredentials Credentials { get; set; }
 
-        public RestClient()
+        private readonly Action _timeoutDelegate;
+
+        public RestClient(Action timeoutDelegate = null)
         {
+            _timeoutDelegate = timeoutDelegate;
```
Comment (scottoffen) [inline src/Grapevine/Client/RestClient.cs]: `RestClient` has to have a parameterless constructor. How about moving this to a public property?

**Comment by kbrimble** [comment]
This should be good to go now.

**Comment by scottoffen** [comment]
Looks good, just two small things:

- 5 of the 6 files in the PR haven't changed, please remove them from your commit.
- Unit test? You had one before, can we put that back in?

Thanks for your help!

**Comment by kbrimble** [comment]
I'm going to close this PR and make a new one tomorrow that's cleaner.

**Pull request closed without merging**